Feature/combobox show add new value while autocompleting

it-vegard opened the pull request:
### Description

Enable option to add a new value while autocompleting (match searchText instead of value). 
Also highlight matches in FilteredOptions to better explain why the options is shown.

### Component Checklist 📝

- [ ] JSDoc
- [ ] Examples
- [ ] Documentation
- [ ] Storybook
- [ ] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [ ] Component tokens (`@navikt/core/css/tokens.json`)
- [ ] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [ ] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [ ] New component? CSS import (`@navikt/core/css/index.css`)
- [ ] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

it-vegard committed:
Show AddNewValue button in Combobox based on searchTerm instead of value, so it is displayed when autocompleting words

it-vegard committed:
Highlight matching text in FilteredOptions

it-vegard committed:
Add changeset

it-vegard committed:
Move styling to combobox.css instead of inline JS

it-vegard committed:
Merge branch 'main' into feature/combobox-show-addNewValue-while-autocompleting

HalvorHaugan commented on .changeset/silver-rings-care.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@navikt/ds-react": patch
```
Should this be a `minor`?

it-vegard replied:
Agreed. Changed it.

it-vegard committed:
Ensure label options are read as one text, even when highlighting search matches.

This only seemed to be a problem in testing library, so we should be able to remove this if they fix this in their calculation of accessible name.

it-vegard committed:
Merge branch 'main' into feature/combobox-show-addNewValue-while-autocompleting

it-vegard committed:
Update changeset to a minor version

HalvorHaugan commented on @navikt/core/react/src/form/combobox/combobox.stories.tsx:
```diff
@@ -186,8 +185,6 @@ export const WithAddNewOptions: StoryFn = ({ open }: { open?: boolean }) => {
       allowNewValues={true}
       shouldAutocomplete={true}
-      value={value}
-      onChange={setValue}
```
I think removal of `value` here and in MultiSelectWithAddNewOptions "breaks" the Chromatic story.

it-vegard committed:
Re-introduce controlled value in storybook examples to not change Chromatic snapshots
